robot_interaction/src/robot_interaction.cpp

```diff
@@ -116,9 +116,28 @@ void RobotInteraction::addActiveComponent(const InteractiveMarkerConstructorFn &
   active_generic_.push_back(g);
 }
 
+static const double DEFAULT_SCALE = 0.25;
+double RobotInteraction::computeLinkMarkerSize(const std::string &group, const std::string &link)
+{
+  const robot_model::JointModelGroup *jmg = robot_model_->getJointModelGroup(group);
+  if (!jmg) return DEFAULT_SCALE;
+
+  robot_state::RobotState default_state(robot_model_);
+  default_state.setToDefaultValues();
+
+  const robot_model::LinkModel *lm = default_state.getLinkModel(link);
+  if (!lm) return DEFAULT_SCALE;
+
+  const Eigen::Vector3d &ext = lm->getShapeExtentsAtOrigin();
+  // slightly bigger than the link diameter
+  double s = 1.01 * ext.norm();
+
+  // clip scale to 5cm
+  return std::max(0.05, s);
```

Comment:
I would agree, that the code should simply rely on its measurements, regardless of magnitude.
However, I didn't invent the 5cm limit. It was there before (see below) and I didn't want to break existing code.
Probably it was there for some reason.